### PR TITLE
AWS documentation fixes

### LINKED ISSUE #698: Document the fact that access to cloud metadata is disabled by default
It's not immediately obvious that `iptables` are overridden with an init container, blocking access to EC2 metadata.

This should be documented somewhere. Perhaps in https://zero-to-jupyterhub.readthedocs.io/en/latest/amazon/step-zero-aws.html.

### PR DESCRIPTION
**Pull request by metonymic-smokey**
Fixes #698 
It contains a brief note about EC2 instance metadata and aligns code blocks on the same page with commands. It also has some minor additions to conf.py.

### CODE REVIEW

**Comment by metonymic-smokey** [comment]
@consideRatio this is a draft PR for #698. Would love your review!

Thanks!

**Comment by betatim** [comment]
Thanks for working on this and helping keep the docs up-to-date and good looking!

I added the `[WIP]` tag to the title of the PR to indicate that it is still a draft. Once you are ready for it to be reviewed thoroughly and then merged you can change it to `[MRG]`.

File: doc/source/conf.py [hunk 1]
```diff
@@ -112,6 +112,26 @@ def setup(app):
...
 
+html_theme_options = {
+    "external_links": [
+        {
+            'url': "https://pandas.pydata.org/pandas-docs/stable/",
+            "name": "Pandas Docs"
+        }
+    ],
+    "github_url": "https://github.com/pandas-dev/pandas-sphinx-theme",
+    "twitter_url": "https://twitter.com/pandas_dev",
```
Comment (betatim) [inline doc/source/conf.py]: Why did you add these three links to the pandas project? I think the github URL one would be useful but it should point to this repository instead.

Not sure we need a link to twitter, it uses up space we could use for something else.

File: doc/source/conf.py [hunk 2]
```diff
@@ -112,6 +112,26 @@ def setup(app):
...
+    "use_edit_page_button": True
```
Comment (betatim) [inline doc/source/conf.py]: It would be great to get the edit on github button back into action. Do you know where it should appear on the page? I looked at the [docs preview](https://856-87849371-gh.circle-artifacts.com/0/html/amazon/step-zero-aws.html) but couldn't find it :(

Comment (manics) [reply doc/source/conf.py]: It's an unresolved issue: https://github.com/pandas-dev/pydata-bootstrap-sphinx-theme/issues/62
I think it'd be best to leave out this config change until the mentioned issue is resolved.

File: doc/source/conf.py (hunk 1 above)
Comment (metonymic-smokey) [reply doc/source/conf.py]: @betatim apologies for the oversight. Will correct this!

File: doc/source/conf.py (hunk 2 above)
Comment (metonymic-smokey) [reply doc/source/conf.py]: @betatim @manics this was brought up in issue #1547. 
Regarding fixing the edit button, there is a temporary fix for it: changing `doc` to `doc/source`. I could add that to this PR if needed?

Thanks for the review!

Comment (manics) [reply doc/source/conf.py]: Looks there's no decision on #1547. If you want to suggest this change anyway could you do it in a separate PR, so it's easy to see the changes and revert them when the theme is fixed?

Comment (metonymic-smokey) [reply doc/source/conf.py]: @manics alright. Shall stick to fixing the other links for now. Will open a separate PR shortly.

Apologies for this, won't try to fix disparate issues in 1 PR now on.

Thanks!

**Comment by metonymic-smokey** [comment]
@betatim are there any more changes you'd like to add before it LGTM?

Thanks!

**Comment by consideRatio** [comment]
@metonymic-smokey thank you and sorry for the extremely slow turnaround to get this merged! :heart: :tada:
